### ISSUE TITLE
GPU delegate: DIV with identical inputs and REDUCE_MAX with const inputs split a small perceiver-style block into 3 partitions (2.1.6)

### ISSUE THREAD

**Issue by john-rocky** [google-ai-edge/LiteRT#9192]
## Summary

Two small GPU-delegate limitations show up together in perceiver-style modules, where a learned latent bank is a constant and attention is taken over a length-1 axis. On litert **2.1.6** (Pixel 8a, Mali-G610, `LITERT_CL`) a 7-node graph splits into **3 partitions, 4 ops on GPU and 3 on CPU**, with:

```
ERROR: Following operations are not supported by GPU delegate:
DIV: No support of few identical inputs in the same operation.
REDUCE_MAX: Expected 1 const input tensor(s), but node has 2 const input(s).
VERBOSE: Replacing 4 out of 7 node(s) with delegate (LITERT_CL) node, yielding 3 partitions for subgraph 0.
```

The graph is numerically fine — this is a residency issue, not a correctness one. I am reporting it because a 3-way partition on a small block costs more than the block itself, and because `DIV` with the same tensor on both inputs looks like something the delegate could accept rather than reject.

## Minimal repro

A softmax over a length-1 axis of a constant buffer. This is what a perceiver-style attention over a single latent position lowers to:

```python
import torch, torch.nn as nn, litert_torch

class ConstSoftmax1(nn.Module):
    def __init__(self):
        super().__init__()
        self.register_buffer("c", torch.randn(1, 4, 1, 32))
    def forward(self, x):
        return torch.softmax(self.c, dim=2) * x   # dim=2 has length 1

m = litert_torch.convert(ConstSoftmax1().eval(), (torch.randn(1, 4, 1, 32),))
open("const_softmax1.tflite", "wb").write(m.model_content())
```

Op set after conversion: `REDUCE_MAX, MAXIMUM, RESHAPE, SUB, EXP, DIV, MUL`. Compiling it with `CompiledModel` and the GPU accelerator produces the log above; CPU runs it without complaint.

The two rejections are independent:

- **`DIV`** — softmax over a length-1 axis makes numerator and denominator the *same* tensor (`exp(0)/exp(0)`), and the delegate declines an op whose two inputs are the same tensor.
- **`REDUCE_MAX`** — its inputs are both constant (the buffer and the axis), and the parser expects at most one constant input.

## Versions

- litert 2.1.6 (current release, pip `ai-edge-litert`): reproduces as above.
- Device: Pixel 8a, Mali-G610, `LITERT_CL`, `CompiledModel`, default fp16 GPU compute.

## Workarounds that work today

Both are model-side and exact:

- Special-case the degenerate axis: a softmax over a length-1 axis is identically 1, so replacing it with `torch.ones_like` removes the whole chain.
- Make one input non-constant (`c + 0.0 * x.mean()`) so the constant-input check no longer applies.

## One observation, in case it is useful

The related pattern — LayerNorm over a constant buffer, which used to hit `MEAN: Expected 1 const input tensor(s)` — no longer reaches the delegate at all on the current toolchain: `litert-torch` constant-folds the whole normalization to a single `MUL`. So for that shape the issue is already resolved upstream of the delegate. I mention it only because it suggests the softmax-over-a-constant case may be foldable in the same place, if that turns out to be a more natural spot to handle it than the delegate's parsers. Happy to test whatever direction you prefer on the device.

**Comment by aravindmurali22** [comment]
Hi @fengwuyao , following up on this thread, is there any recent updates on this? Thanks!
